CMakeLists.txt

```diff
@@ -215,15 +215,22 @@ if(LINUX)
   set(HAVE_UDEV ${UDEV_FOUND})
   find_package(blkid REQUIRED)
   set(HAVE_BLKID ${BLKID_FOUND})
+  if(WITH_RBD)
+    find_package(genl REQUIRED)
+    set(HAVE_GENL $GENL_FOUND)
```

Comment:
I guess technically the new code should be wrapped in ```#if HAVE_GENL``` clauses (and update ```src/include/config-h.in.cmake```)?

Reply:
Ah, I did a double mess up maybe :)

The WITH_RBD check above is always false, so the:

find_package(genl REQUIRED)

is just never running. Because it's "required" I do not think we need to add HAVE_GENL ifdefs to the code since cmake will fail if the netlink libs are not installed.

However, when I respin this patch do you want me to relax that requirement, make it optional and then add ifdefs in the code?

Reply:
If all the OSs we support for Nautilus have the package (EL7, EL8, Xenial, Bionic, Buster, OpenSUSE, etc), no need to add ```#ifdef```s all over the place.

Reply:
The netlink libs are in all those distros. I broke out the build, cmake, control parts to a separate patch in this same PR:

https://github.com/ceph/ceph/pull/27902/commits/88156715536924293fb3a8b610566369324e69db

to make it easier to check.

I tested the RHEL variants and checked the rpm names were the same in fedora, Xenial (only checked that the packages were the same name in Bionic and Buster), and the current open suse 15.* (I did not test the versions marked EOL here https://en.wikipedia.org/wiki/OpenSUSE_version_history and only checked that they did have the packages and the names didn't change).